ADR/xxxx-kite-poc-architecture-and-components.md

```diff
@@ -0,0 +1,244 @@
+        DASH[Issues Dashboard]
+        CLI[KITE CLI Tool]
+        EXT[External Tools:<br/> Monitoring, Alerts]
+    end
+
+    %% Operator Flow
+    BO -->|watches| K8S
+    BO -->|watches specific resource| CTRL1
+    BO -->|watches specific resource| CTRL2
+    BO -->|watches specific resource| CTRL3
+    CTRL1 -->|HTTP POST| API
+    CTRL2 -->|HTTP POST| API
+    CTRL3 -->|HTTP POST| API
+
+    %% Backend Flow
+    API -->|handle issue creation or update| SVC
+    WH -->|handle issue creation or update| SVC
+    SVC -->|Duplicate Prevention| REPO
+    REPO -->|ACID Transactions| PG
+
+    %% User Interface Flow
+    DASH -->|Query Issues| API
+    CLI -->|Query Issues| API
+    EXT -->|REST API| API
+```
+
+## Decision
+
+We will implement KITE as a distributed system with the following key architectural decisions:
+
+### Bridge Operator Architecture
+
+The **[KITE Bridge Operator](https://github.com/konflux-ci/kite/tree/main/packages/operator)** implements the "bridge operator" pattern, which connects a Kubernetes environment with external systems not natively managed by Kubernetes.
+
+The operator:
+
+- **Monitors Kubernetes Resources**: Watches for events on various cluster resources (currently focusing on Tekton `PipelineRun` objects)
+- **Detects State Changes**: Identifies successes or failures in monitored resources
+- **Reports to Backend**: Sends failure information to the KITE backend service for persistence
+- **Extensible Design**: Can be extended with additional controllers for monitoring other resource types
+
+The operator runs as a standard Kubernetes controller with cluster-wide permissions to monitor resources across namespaces.
+
+### KITE Backend Service
+
+The **[KITE Backend](https://github.com/konflux-ci/kite/tree/main/packages/backend)** is a Go-based REST API service that:
+
+- **Provides API Endpoints**: Offers [REST API](https://github.com/konflux-ci/kite/blob/main/packages/backend/docs/API.md) for creating, updating, and querying issues
+- **Webhook Support**: Includes [specialized webhook endpoints](https://github.com/konflux-ci/kite/blob/main/packages/backend/docs/Webhooks.md) for simplified issue creation/resolution
+- **Issue Management**: Handles the complete lifecycle of issues (creation, updates, resolution)
+- **Database Integration**: Manages all database operations and data persistence
+- **Namespace isolation**: Issue access is namespace-restricted for isolation and security. (WIP)
+
+The backend is built using the [Gin](https://github.com/gin-gonic/gin) HTTP web framework and follows standard HTTP API patterns.
+
+### Team Integration Strategy
+
+KITE provides two primary integration paths for teams to onboard their services and start tracking issues:
+
+#### Recommended Integration Path
+
+**1. Build Custom Controllers**
+Teams [develop custom controllers](https://github.com/konflux-ci/kite/blob/main/packages/operator/docs/ControllerDevelopmentGuide.md) for the KITE Bridge Operator that:
```

Comment:
So all this service-specific knowledge would have to live in the Kite repo instead of the service repo?

Reply:
I would imagine that the custom controllers would have to live within the Kite repository, unless there is some way to dynamically add a controller to an operator? Also which service repo are you referring to?